src/payment/models/buy-crypto/services/buy-crypto.service.ts

```diff
@@ -93,11 +90,13 @@ export class BuyCryptoService {
       if (!update.chargebackBankTx) throw new BadRequestException('Bank TX not found');
     }
 
-    // buy
-    if (dto.buyId) update.buy = await this.getBuy(dto.buyId);
-
-    // crypto route
-    if (dto.cryptoRouteId) update.cryptoRoute = await this.getCryptoRoute(dto.cryptoRouteId);
+    if (entity.bankTx) {
+      // buy
+      if (dto.buyId) update.buy = await this.getBuy(dto.buyId);
+    } else {
+      // crypto route
+      if (dto.cryptoRouteId) update.cryptoRoute = await this.getCryptoRoute(dto.cryptoRouteId);
+    }
```

Comment:
I would throw an error, if somebody tries to update buy, but entity does not have a buy yet. Something like this:
```ts
if (dto.buyId) {
   if (!entity.buy) throw new BadRequestException(`Cannot assign BuyCrypto ${id} to a buy route`);
   update.buy = await this.getBuy(dto.buyId);
}
```

What do you think?